source/extensions/common/redis/cluster_refresh_manager_impl.cc

```diff
@@ -90,6 +90,11 @@ bool ClusterRefreshManagerImpl::onEvent(const std::string& cluster_name, EventTy
           }
         });
         return true;
+      } else if (info->last_callback_time_ms_.load() != last_callback_time_ms) {
```

Comment:
If we stick with this approach I think it would be more clear if this was a basic else statement paired with the CAS, inside a large if statement with the counter increment.

Reply:
I think if we have the if counter set to 0 is in a if statement, it would also race. because if thread 1 set the counter to 0, the incr in thread 2 might not trigger the threshold.  Let me rewrite this logic to make it clearer.